### ISSUE TITLE
Level editor view movement

### ISSUE THREAD

**Issue by raptor** [bitfighter/bitfighter#187]
_From [Jomskylark](https://code.google.com/u/Jomskylark/) on December 19, 2012 19:12:05_

Moving the level editor's viewpoint diagonally via WASD is only possible if W (up) or S (down) is pressed, THEN the left/right key. However, pressing A (left) or D (right) BEFORE W or S negates the up/down movement.

Not a big deal but it makes the level editor movement feel "glitchy."

_Original issue: http://code.google.com/p/bitfighter/issues/detail?id=187_

**Comment by raptor** [comment]
_From [buckyballreaction](https://code.google.com/u/buckyballreaction/) on December 24, 2012 10:18:00_

watusimoto, this is because we're testing for input strings instead of input codes in EditorUserInterface::onKeyDown().  When pressing 'A' + 'S', the inputString returned is just 'A' (see diagnostics).

The solution is to test for input codes again.  What was the reason for moving to input strings in the first place?  I don't remember..

**Comment by raptor** [comment]
_From [watusim...@bitfighter.org](https://code.google.com/u/105427273526970468779/) on December 30, 2012 03:21:03_

Correct diagnosis, but wrong solution, as raw input codes cannot differentiate between Up and Shift+Up (for example).  The solution I implemented looks at what keys you've most recently pressed, and selects that one if more than one key is pressed.  Previously, we focused on an arbitrary key.

Checkin 6313 fixes this issue.

**Status:** Fixed
